src/client/transport.js

```diff
@@ -1,245 +1,240 @@
 /*eslint-disable no-native-reassign*/
-import * as JSON from './json';
 import NativeMethods from './sandbox/native-methods';
-import ServiceCommands from '../service-msg-cmd';
-import * as Browser from './utils/browser';
 import EventEmitter from './utils/event-emitter';
 import Settings from './settings';
+import { stringify as stringifyJSON, parse as parseJSON } from './json';
+import { setCookie as SET_COOKIE_CMD } from '../service-msg-cmd';
+import { isWebKit, isIE9 } from './utils/browser';
+import { Promise } from 'es6-promise';
 
 /*eslint-enable no-native-reassign*/
 
-var Promise = require('es6-promise').Promise;
+class Transport {
+    constructor () {
+        this.SWITCH_BACK_TO_ASYNC_XHR_DELAY    = 2000;
+        this.SERVICE_MESSAGES_WAITING_INTERVAL = 50;
+        this.MSG_RECEIVED                      = 'received';
 
-//Const
-var SWITCH_BACK_TO_ASYNC_XHR_DELAY    = 2000;
-var SERVICE_MESSAGES_WAITING_INTERVAL = 50;
-var MSG_RECEIVED                      = 'received';
+        this.eventEmitter                 = new EventEmitter();
+        this.msgQueue                     = {};
+        this.useAsyncXhr                  = true;
+        this.activeServiceMessagesCounter = 0;
 
-//Globals
-var eventEmitter                 = new EventEmitter();
-var msgQueue                     = {};
-var useAsyncXhr                  = true;
-var activeServiceMessagesCounter = 0;
-var Transport                    = {};
+        //NOTE: if we are unloading we should switch to sync XHR to be sure that we will not lost any service msgs
+        window.addEventListener('beforeunload', () => {
+            this.useAsyncXhr = false;
 
-//NOTE: if we are unloading we should switch to sync XHR to be sure that we will not lost any service msgs
-window.addEventListener('beforeunload', function () {
-    useAsyncXhr = false;
-
-    //NOTE: if unloading was canceled switch back to async XHR
-    NativeMethods.setTimeout.call(window, function () {
-        useAsyncXhr = true;
-    }, SWITCH_BACK_TO_ASYNC_XHR_DELAY);
-}, true);
+            //NOTE: if unloading was canceled switch back to async XHR
+            NativeMethods.setTimeout.call(window, () => this.useAsyncXhr = true, this.SWITCH_BACK_TO_ASYNC_XHR_DELAY);
+        }, true);
+    }
 
-Transport.sendNextQueuedMsg = function (queueId) {
-    var queueItem = msgQueue[queueId][0];
+    static _createXMLHttpRequest (async) {
+        var xhr = new NativeMethods.XMLHttpRequest();
```

Comment:
We have a `SERVICE_MSG_REQUEST_FLAG` in https://github.com/churkin/testcafe-hammerhead/blob/es6/src/client/sandbox/xhr/index.js . It was used to skip tracking of the transport messages. I'm wondering, since we use native methods now, do we still need this flag?